src/sip/auth.c

```diff
@@ -81,30 +87,50 @@ static void auth_destructor(void *arg)
 static int mkdigest(uint8_t *digest, const struct realm *realm,
 		    const char *met, const char *uri, uint64_t cnonce)
 {
-	uint8_t ha1[MD5_SIZE], ha2[MD5_SIZE];
+	uint8_t *ha1, *ha2;
+	int h_size;
 	int err;
+	digest_printf_h *digest_printf;
+	bool use_sha256 = str_casecmp(realm->algorithm, "sha-256") == 0;
+
+#ifndef USE_OPENSSL
+	if (use_sha256) {
+		DEBUG_WARNING("SHA2 digest only supported "
+				"when compiled with OpenSSL\n");
+		return 1;
+	}
+#endif
+	h_size = use_sha256 ? SHA256_SIZE : MD5_SIZE;
+	ha1 = mem_zalloc(h_size, NULL);
+	ha2 = mem_zalloc(h_size, NULL);
```

Comment:
It looks like for ha1 and ha2 there is no mem_deref?

Reply:
good catch - also `digest` is missing a mem_deref; I'll add them soon, thanks.

Reply:
pushed